src/main/java/intellistart/interviewplanning/controllers/dto/DashboardMapDto.kt

```diff
@@ -0,0 +1,73 @@
+package intellistart.interviewplanning.controllers.dto
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import intellistart.interviewplanning.model.candidateslot.CandidateSlot
+import intellistart.interviewplanning.model.dayofweek.DayOfWeek
+import intellistart.interviewplanning.model.interviewerslot.InterviewerSlot
+import intellistart.interviewplanning.model.week.WeekService
+import java.time.LocalDate
+import java.util.*
+
+/**
+ * Dto object for representation all candidate, interviewer slots
+ * and bookings for a certain week.
+ */
+data class DashboardMapDto(
+    @JsonIgnore
+    val weekService: WeekService,
+    val weekNum: Long,
+    val dashboard: HashMap<LocalDate, DashboardDto>
+) {
+    /**
+     * Constructor to initialize object with given weekNum.
+     * Needs [WeekService] instance to access weeks logic
+     * (gaining date from week number and day, etc).
+     *
+     * @param weekNum number of week
+     * @param weekService instance of service object to access logic from
+     */
+    constructor(weekNum: Long, weekService: WeekService) : this(
+        weekService,
+        weekNum,
+        HashMap<LocalDate, DashboardDto>().apply {
+            DayOfWeek.entries.forEach { dayOfWeek ->
+                val date = weekService.convertToLocalDate(weekNum, dayOfWeek)
+                put(date, DashboardDto(HashSet(), HashSet(), HashMap()))
```

Comment:
```suggestion
                put(date, DashboardDto(emptySet(), emtySet(), emptyMap()))
```

Reply:
Collection data is later populated, so it cannot be created by emptySet() and emptyMap()